test/e2e/storage/testsuites/testdriver.go

```diff
@@ -99,9 +99,42 @@ type DriverInfo struct {
 	RequiredMountOption  sets.String         // Map of string for required mount option (Optional)
 	Capabilities         map[Capability]bool // Map that represents plugin capabilities
 
-	// Parameters below will be set inside test loop by using SetCommonDriverParameters.
-	// Drivers that implement TestDriver is required to set all the above parameters
-	// and return DriverInfo on GetDriverInfo() call.
-	Framework *framework.Framework       // Framework for the test
-	Config    framework.VolumeTestConfig // VolumeTestConfig for thet test
+	Config TestConfig // Test configuration for the current test.
+}
+
+// TestConfig represents parameters that control test execution.
+// They can still be modified after defining tests, for example
+// in a BeforeEach or when creating the driver.
+type TestConfig struct {
```

Comment:
The existence of two `TestConfig`s is going to be very confusing. Is it possible to have a clearer divide between the purposes of the two structs?

Some possibilities:
* Improve the comments for `VolumeTestConfig` clearly describing its purpose, and possibly renaming it.
  * Most fields in `VolumeTestConfig` are only used in `volume_util.go`. Could we limit its use only in that file?
* Reduce (or ideally completely eliminate) field overlap between the two structs.

Reply:
As far as I understand it, `VolumeTestConfig` is from legacy code that @mkimuram continued to use unmodified because some tests and utility code rely on it. I agree that it would be useful to clean that up, but I barely know that code and would prefer to not touch it, or at least not now.

I'll try to enhance the documentation for the new struct, but when it comes to `VolumeTestConfig` I'll basically have to tell developers that they'll have to read the legacy code.

Reply:
See 0fa0ac7.

Reply:
OK, I could create an issue to track the cleanup between the two configs in the future.

Reply:
https://github.com/kubernetes/kubernetes/issues/72242